loadtest/evm.go

```diff
@@ -65,35 +72,83 @@ func NewEvmTxClient(
 	return txClient
 }
 
-// GenerateEvmSignedTx takes a private key and generate a signed bank send TX
+func (txClient *EvmTxClient) GetTxForMsgType(msgType string) *ethtypes.Transaction {
+	switch msgType {
+	case EVM:
+		return txClient.GenerateSendFundsTx()
+	case ERC20:
+		return txClient.GenerateERC20TransferTx()
+	default:
+		panic("invalid message type")
+	}
+}
+
+func randomValue() *big.Int {
+	return big.NewInt(rand.Int63n(9000000) * 1000000000000)
+}
+
+// GenerateSendFundsTx returns a random send funds tx
 //
 //nolint:staticcheck
-func (txClient *EvmTxClient) GenerateEvmSignedTx() *ethtypes.Transaction {
-	txClient.mtx.RLock()
-	defer txClient.mtx.RUnlock()
+func (txClient *EvmTxClient) GenerateSendFundsTx() *ethtypes.Transaction {
+	tx := ethtypes.NewTx(&ethtypes.LegacyTx{
+		Nonce:    txClient.nextNonce(),
+		GasPrice: txClient.gasPrice,
+		Gas:      uint64(21000),
+		To:       &txClient.accountAddress,
+		Value:    randomValue(),
+	})
+	return txClient.sign(tx)
+}
 
-	privKeyHex := hex.EncodeToString(txClient.privateKey.Bytes())
-	privateKey, err := crypto.HexToECDSA(privKeyHex)
+// GenerateERC20TransferTx returns a random ERC20 send
+// the contract it interacts with needs no funding (infinite balances)
+func (txClient *EvmTxClient) GenerateERC20TransferTx() *ethtypes.Transaction {
+	opts := txClient.getTransactOpts()
+	// override gas limit for an ERC20 transfer
+	opts.GasLimit = uint64(100000)
+	tokenAddress := txClient.evmAddresses.ERC20
+	token, err := erc20.NewErc20(tokenAddress, GetNextEthClient(txClient.ethClients))
 	if err != nil {
-		fmt.Printf("Failed to load private key: %v \n", err)
+		panic(fmt.Sprintf("Failed to create ERC20 contract: %v \n", err))
 	}
+	tx, err := token.Transfer(opts, txClient.accountAddress, randomValue())
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create ERC20 transfer: %v \n", err))
+	}
+	return tx
+}
 
-	// Get the next nonce
-	nextNonce := txClient.nonce.Add(1) - 1
+func (txClient *EvmTxClient) getTransactOpts() *bind.TransactOpts {
+	auth, err := bind.NewKeyedTransactorWithChainID(txClient.privateKey, txClient.chainId)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create transactor: %v \n", err))
+	}
+	auth.Nonce = big.NewInt(int64(txClient.nextNonce()))
+	auth.Value = big.NewInt(0)
+	auth.GasLimit = uint64(21000)
+	auth.GasPrice = txClient.gasPrice
+	auth.Context = context.Background()
+	auth.From = txClient.accountAddress
+	auth.NoSend = true
```

Comment:
hm why is this true? Are we trying to not send it for some reason?

Reply:
this prevents the abigen bindings from auto-sending it, so that the load test client can send the signed one later